webapp/components/test-file-results.html

```diff
@@ -97,33 +91,43 @@
         if (!path || !testRuns) {
           return;
         }
-        const resultPerTestRun = await Promise.all(
+        const resultsPerTestRun = await Promise.all(
           testRuns.map(tr => this.loadResultFile(tr)));
 
-        const subtestNames = ['STATUS'];
-        let hasSubtests = false;
-        resultPerTestRun.forEach((resultData, i) => {
-          if (!resultData) {
-            testRuns[i].subtests = {};
-            testRuns[i].subtests['STATUS'] = { status: '(results not found)' };
-            return;
-          }
-          testRuns[i].subtests = {};
-          testRuns[i].subtests['STATUS'] = { status: resultData.status };
-          if (resultData.subtests && resultData.subtests.length > 0) {
-            // Since we spoof the file-level status also as a "subtest", we
-            // store whether the test originally has subtests here.
-            hasSubtests = true;
-            for (let subtestResult of resultData.subtests) {
-              testRuns[i].subtests[subtestResult.name] = subtestResult;
-              if (!subtestNames.includes(subtestResult.name)) {
-                subtestNames.push(subtestResult.name);
-              }
-            }
-          }
-        });
-        this.hasSubtests = hasSubtests;
-        this.subtestNames = subtestNames;
+        // resultsTable[0].name set after discovering subtests.
+        let resultsTable = [{
+          results: resultsPerTestRun.map(data => {
+            return {status: data && data.status};
+          }),
+        }];
+
+        let subtestNamesSet = new Set();
```

Comment:
Is Javascript `Set` ordered? If not, this changes the behaviour, i.e. the order of subtests is no longer preserved. Although the current behaviour (sorting based on the first test run) isn't perfect either, I do think it's better than not sorting subtests at all. For tests like idlharness.js tests, we do want to try our best to preserve the ordering.

I guess this can be changed to a simple array, which will increase the complexity of the insert-if-not-exist operation below to O(N^2), but I think that's fine.